app/src/main/java/com/babylon/wallet/android/presentation/main/MainViewModel.kt

```diff
@@ -94,7 +94,6 @@ class MainViewModel @Inject constructor(
                 else -> p2PLinksRepository.observeP2PLinks().drop(1)
             }
         }
-        .distinctUntilChanged()
```

Comment:
the only bad thing is that now it will try to connect for every change in the profile, but I don't think there is other way (with this implementation)

Reply:
There is also an overload method for this which lets you decide if the values have changed between the `old` and `new`. Do you think this would help?

Reply:
what is this overload method? 👀

Reply:
That's right, the connection code gets unnecessarily called here, but the good thing is that PeerdroidConnector won't attempt reconnecting when the connection already exists. The same is true for the dapp incoming requests listening, the related code won't be unnecessarily executed.